core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -463,82 +466,100 @@ public void transportTerminated() {
     @Override
     public void streamCreated(
         final ServerStream stream, final String methodName, final Metadata headers) {
-      if (headers.containsKey(MESSAGE_ENCODING_KEY)) {
-        String encoding = headers.get(MESSAGE_ENCODING_KEY);
-        Decompressor decompressor = decompressorRegistry.lookupDecompressor(encoding);
-        if (decompressor == null) {
-          stream.close(
-              Status.UNIMPLEMENTED.withDescription(
-                  String.format("Can't find decompressor for %s", encoding)),
-              new Metadata());
-          return;
+      final Tag tag = PerfMark.createTag(methodName, stream.hashCode());
+      PerfMark.startTask("ServerTransportListener.streamCreated", tag);
+      try {
```

Comment:
This is a lot to put into the try-finally. Could we pass tag as an argument to a function instead?

Would probably be a good idea for ClientStreamListener.closed as well. And for some of the Netty changes.

Reply:
Done.